classes/Link.php

```diff
@@ -820,6 +824,30 @@ public function getAdminLink($controller, $withToken = true, $sfRouteParams = ar
         return $this->getAdminBaseLink() . basename(_PS_ADMIN_DIR_) . '/' . Dispatcher::getInstance()->createUrl($controller, $idLang, $params);
     }
 
+    /**
+     * @param \Symfony\Component\Routing\RouterInterface $sfRouter
+     * @param string                                     $controller
+     *
+     * @return string|null
```

Comment:
When we can do otherwise, please returns only 1 type: here you can return an empty string...
but is it a good idea to return an empty string or null? Shouldn't we throw an exception if the route is not found?

Reply:
ok I can return null, I personaly prefer to check nullity rather than empty strings but it's a matter of tastes I don't mind changing it
about the exception I'm not sure, because this route search is optional and has few chances to return a result (only ps_linklist manages it for now, for ONE route)
throwing an exception would mean adding an empty try catch block in the calling method, I don't see the point in this case